internal/pkg/template/workload.go

```diff
@@ -290,14 +284,39 @@ func (cfg *NetworkLoadBalancer) Aliases() []string {
 	for _, listener := range cfg.Listener {
 		for _, entry := range listener.Aliases {
 			if _, value := seen[entry]; !value {
-				seen[entry] = true
-				uniqueAliases = append(uniqueAliases, entry)
+				uniqueAliases = append(uniqueAliases, uniqueEntriesFromList(listener.Aliases, seen)...)
```

Comment:
Maybe I misunderstood something but I am not following why we are doing this. It seems like we are not appending `entry` which is a unique alias. Instead, we do `append(uniqueAliases, uniqueEntriesFromList(listener.Aliases, seen)...)` multiple times whenever we see any alias that is not in `seen` map.

Reply:
So there are couple of things happening:
1. In the previous PR, we decided to remove `uniqueEntriesFromList()` method and add that logic inside `Aliases()` method itself. But now for ALB also we need the similar kind of logic so to remove duplication of code I removed common code out in the `uniqueEntriesFromList()` method which nothing but just gives the non-duplicate entries by checking the map that we fill as we get aliases from different listener rules. 
2. Consider following example with multiple additional rules
```
Listener: []ApplicationLoadBalancerRoutineRule{
					{
						Aliases: []string{
							"testAlias1",
							"testAlias2",
						},
					},
					{
						Aliases: []string{
							"testAlias1",
							"testAlias3",
						},
					},
				},
```
From the example you can see that listener1 has aliases as "testAlias1" and "testAlias2" whereas listener2 has "testAlias1" and "testAlias3". So in order to create certificate for all the unique aliases across all the listeners we need to write logic that traverses through all the listeners which is happening in `Aliases()` method and we are figure our all the aliases that are unique across all the listeners which happens in `uniqueEntriesFromList()`. 
Does that make sense?

Reply:
I am still not quite following. I feel like the original version is doing exactly the work and will return `[]string{testAlias1, testAlias2, testAlias3}` 🤔 Your modification will pass the unit tests but seems redundant or can you give me an example that this function previously won't work?

Reply:
`uniqueEntriesFromList(listener.Aliases, seen)...)` is just added to remove the duplication of code from `Aliases()` method that is there for BackendService and LoadBalancedService. If you want I can keep the code as is, but there will code duplication for both LBWS and BS.

Reply:
```golang
uniqueAliases = append(uniqueAliases, entry)
seen[entry] = true
```
does this not work?